runtime-light/streams/stream.h

```diff
@@ -37,9 +37,7 @@ class stream {
       : m_storage(std::move(storage)),
         m_storage_capacity(capacity),
         m_descriptor(descriptor) {
-    if (m_storage_capacity != 0) {
-      kphp::log::assertion(static_cast<bool>(m_storage));
-    }
+    kphp::log::assertion(static_cast<bool>(m_storage_capacity == 0 || m_storage));
```

Comment:
Maybe rewrite it like `m_storage_capacity == 0 || static_cast<bool>(m_storage) `